src/app/public/modules/ag-grid/cell-renderers/cell-renderer-currency/cell-renderer-currency-validator.component.ts

```diff
@@ -0,0 +1,31 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { ICellRendererAngularComp } from 'ag-grid-angular';
+import { SkyCellRendererCurrencyParams } from '../../types/cell-renderer-currency-params';
+
+@Component({
+  selector: 'sky-ag-grid-cell-renderer-currency-validator',
+  templateUrl: 'cell-renderer-currency-validator.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyAgGridCellRendererCurrencyValidatorComponent implements ICellRendererAngularComp {
```

Comment:
I'm curious if we need this validator - or if we can just simply use branching logic in the current renderer to achieve this without having a separate renderer. Then consumers would just need to pass in a validation function and message when they want to validate and not pass it in otherwise.

Reply:
Happy to chat about this if you'd like.

Reply:
I did a lot to try to consolidate the two, but it also involves enabling/disabling the cell event listeners and it was easier to have AG Grid choose between two component types.